mscore/palettebox.cpp

```diff
@@ -63,6 +69,18 @@ PaletteBox::PaletteBox(QWidget* parent)
       hlSearch->setContentsMargins(5,0,5,0);
       hlSearch->addWidget(_searchBox);
 
+
+      QFile file(":/default.txt");
+      file.open(QIODevice::ReadOnly);
+      TreeModel model(file.readAll());
```

Comment:
The model will be deleted as soon as the function (in this case the PaletteBox constructor) goes out of scope, so you will never have a chance to see it. To avoid this, you need to create it as a pointer instead:

```c++
TreeModel* model = new TreeModel; // creates a pointer called "model" that points to a TreeModel object
```

Now only the pointer will be destroyed, not the object it points to. However, this leads to a new problem: the object will *never* be destroyed! If you keep creating objects this way then you would eventually run out of memory. To avoid this, you usually have to remember to tidy up later by calling:

```c++
delete model; // this deletes both the pointer called "model" and also the object it points to
```

That is an ordinary C++ feature, but since we are using Qt there is an easier method. Simply give the object a "parent" and Qt will delete it for you when its parent is destroyed:

```c++
TreeModel* model = new TreeModel;
model->setParent(parent);
```

There is a shorter method to do the same thing:

```c++
TreeModel* model = new TreeModel(parent);
```

The parent can be a pointer to any Qt object (e.g. a QTreeView) or any MuseScore object that inherits from a Qt object (e.g. PaletteBox, which [inherits from QDockWidget](https://github.com/musescore/MuseScore/blob/55acd81bb68a50ff7dff2338eb22b1c40580d722/mscore/palettebox.h#L28)). Since you are inside the PaletteBox's constructor, you can refer to it using the `this` keyword:

```c++
TreeModel* model = new TreeModel(this);
```

Note: now that `model` is a pointer, you will need to find-and-replace all occurances of `model.` with `model->` to make it compile.